### ISSUE TITLE
Testing for JRDB dataset

### ISSUE THREAD

**Issue by HelloWorld416** [AdaCompNUS/NMRF_TrajectoryPrediction#1]
Thanks author open the code for their excellent work! I wonder if the evaluation results on JRDB dataset in paper are used with the author provided in 
**./processed_datasets** [https://github.com/AdaCompNUS/NMRF_TrajectoryPrediction/blob/main/processed_datasets/jrdb/trajectories_jrdb_val.pkl](url).

**Comment by fangzlinn** [comment]
Hi,
Thank you for your interest, and sorry for the delayed reply, as I did not receive any email notification.

As stated in our paper for the JRDB dataset:

> “We follow the train-validation-test split applied in Social-Transmotion (Saadatnejad et al., 2023) for deterministic prediction. For stochastic situations, splits are in accordance with the official JRDB detection and tracking challenge.”

There are two cases:
*1) Deterministic case*
For the deterministic setting, the original JRDB dataset only provided labels for the train and validation portions when we conducted our experiments. Therefore, the authors of Social-Transmotion manually selected several scenes from the JRDB training set as a validation split. We followed the same setting for deterministic evaluation, but we did not provide separate `.pkl` files for this split. So, to reproduce the deterministic results in Table 4, you would need to: 1) refer to the Social-Transmotion split, 2) manually separate the corresponding validation scenes from `trajectories_jrdb_train.pkl` (or, you can also refer to *jrdb_toolkit* provided by the dataset authors to generate splits), and 3) use `trajectories_jrdb_val.pkl` as the test set.

*2) Stochastic case*
For the stochastic setting (Table 5), we used the official JRDB detection and tracking challenge split. In this case, `trajectories_jrdb_train.pkl` is used for training, and `trajectories_jrdb_val.pkl` is used for evaluation.
